### PR TITLE
Feature/obligation validation

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(1.1)]
+        [InlineData(1.11)]
+        [InlineData(1.111)]
+        [InlineData(" ")]
+        [InlineData("  ")]
+        public void Validate_ValueIsValid_ReturnsTonnageValidationResultSuccess(object input)
+        {
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs:70]: this is invalid so have added an invalid test case

File: src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            var result = validator.Validate(input);
+
+            result.Should().Be(TonnageValidationResult.Success);
+        }
+
+        [Theory]
+        [InlineData("00000000000000")]
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs:77]: added some 13s...

File: src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+            result.Should().Be(TonnageValidationResult.Success);
+        }
+
+        [Theory]
+        [InlineData("000000000000000")]
+        [InlineData("000000000000000.00")]
+        [InlineData("000000000000000.000")]
+        [InlineData("000000000000000.0")]
+        public void Validate_GivenValueHasMoreThanFourteenIntegerParts_ReturnsTonnageValidationResultSuccess(object input)
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core.Tests.Unit/Validation/TonnageValueValidatorTests.cs]: updated

File: src/EA.Weee.Core/Shared/CsvReading/ObligationCsvUpload.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+﻿namespace EA.Weee.Core.Shared.CsvReading
+{
+    using Domain.Lookup;
+
+    public class ObligationCsvUpload
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core/Shared/CsvReading/ObligationCsvUpload.cs:5]: Not at the moment, the CSV handling is all handled in the handler. The csv processed data isn't currently passed by to the client

File: src/EA.Weee.RequestHandlers/Admin/Obligations/ObligationUploadValidator.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        public async Task<IList<ObligationUploadError>> Validate(IList<ObligationCsvUpload> obligations)
+        {
+            //TODO validate PCS belongs to AA
+            var validationErrors = new List<ObligationUploadError>();
+            
+            foreach (var obligationCsvUpload in obligations)
+            {
+                var findScheme =
+                    await schemeDataAccess.GetSchemeOrDefaultByApprovalNumber(
+                        obligationCsvUpload.SchemeIdentifier.Trim());
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/Admin/Obligations/ObligationUploadValidator.cs:33]: The trim is against the data in the CSV data which would be string.empty is there was no entry.

File: src/EA.Weee.Core/Shared/CsvReading/ObligationCsvUpload.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.Core/Shared/CsvReading/ObligationCsvUpload.cs:5]: noted

File: src/EA.Weee.RequestHandlers/Admin/Obligations/ObligationUploadValidator.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (meglenapnovoselska) [reply src/EA.Weee.RequestHandlers/Admin/Obligations/ObligationUploadValidator.cs:33]: alright

**Pull request merged**